0001-data-collection-extension/README.md

```diff
@@ -67,6 +64,77 @@ Device shadow. The user is only interested in the current status of the endpoint
 ### Use case 2
 Historical data collection. The user is interested in storing all collected data with timestamps for further processing and visualizing historical trends.
 
+## Design
+
+### Batch uploads
+To reduce network usage, all records are uploaded in _batches_. All records in a batch are processed as one and have a single response status.
+
+### Timestamps
+Due to the fact different application might need timestamps in different formats and precision, data collection extension does not provide any special handling for timestamps. That's responsibility of higher layers to interpret any field as a timestamp.
+
+#### Absent timestamp handling
+As there is no special field for a timestamp, that's higher level's issue how to handle absent timestamps.
+
+The recommended solution is to save server timestamp upon network message receiving and pass it along the parsed data, so upper layers can use receiving timestamp if needed.
```

Comment:
> I would not—that's not our business.
Yep, you're right.

> it's not per, it's a time when the timestamp is set.

I think, I missed something. Is network message mqtt publish message?